source/common/network/listener_impl.cc

```diff
@@ -43,85 +41,92 @@ bool ListenerImpl::rejectCxOverGlobalLimit() {
+    // `sockaddr_un` associated with the client socket when starting from the server socket.
+    // We work around this by using our own name for the socket in this case.
+    // Pass the 'v6only' parameter as true if the local_address is an IPv6 address. This has no
+    // effect if the socket is a v4 socket, but for v6 sockets this will create an IPv4 remote
+    // address if an IPv4 local_address was created from an IPv6 mapped IPv4 address.
+    const Address::InstanceConstSharedPtr& remote_address =
+        (remote_addr.ss_family == AF_UNIX)
+            ? io_handle->peerAddress()
+            : Address::addressFromSockAddr(remote_addr, remote_addr_len,
+                                           local_address->ip()->version() ==
+                                               Address::IpVersion::v6);
+
+    cb_.onAccept(
+        std::make_unique<AcceptedSocketImpl>(std::move(io_handle), local_address, remote_address));
   }
-
-  // Get the local address from the new socket if the listener is listening on IP ANY
-  // (e.g., 0.0.0.0 for IPv4) (local_address_ is nullptr in this case).
-  const Address::InstanceConstSharedPtr& local_address =
-      listener->local_address_ ? listener->local_address_ : io_handle->localAddress();
-
-  // The accept() call that filled in remote_addr doesn't fill in more than the sa_family field
-  // for Unix domain sockets; apparently there isn't a mechanism in the kernel to get the
-  // `sockaddr_un` associated with the client socket when starting from the server socket.
-  // We work around this by using our own name for the socket in this case.
-  // Pass the 'v6only' parameter as true if the local_address is an IPv6 address. This has no effect
-  // if the socket is a v4 socket, but for v6 sockets this will create an IPv4 remote address if an
-  // IPv4 local_address was created from an IPv6 mapped IPv4 address.
-  const Address::InstanceConstSharedPtr& remote_address =
-      (remote_addr->sa_family == AF_UNIX)
-          ? io_handle->peerAddress()
-          : Address::addressFromSockAddr(*reinterpret_cast<const sockaddr_storage*>(remote_addr),
-                                         remote_addr_len,
-                                         local_address->ip()->version() == Address::IpVersion::v6);
-  listener->cb_.onAccept(
-      std::make_unique<AcceptedSocketImpl>(std::move(io_handle), local_address, remote_address));
 }
 
 void ListenerImpl::setupServerSocket(Event::DispatcherImpl& dispatcher, Socket& socket) {
-  listener_.reset(
-      evconnlistener_new(&dispatcher.base(), listenCallback, this, 0, -1, socket.ioHandle().fd()));
-
-  if (!listener_) {
-    throw CreateListenerException(
-        fmt::format("cannot listen on socket: {}", socket.localAddress()->asString()));
-  }
+#ifdef WIN32
+  auto trigger = Event::FileTriggerType::Level;
+#else
+  auto trigger = Event::FileTriggerType::Edge;
+#endif
+  socket.ioHandle().listen(128);
```

Comment:
Use a constant for the 128 backlog.

What was the listen backlog prior to these changes?  Should the backlog be configurable?

Reply:
See https://github.com/envoyproxy/envoy/issues/9501. @florincoras can you fix that as part of a follow up change? Would really appreciate it. It should be much easier now. In the interim we should replicate what libevent does which IIRC is 128 as you did here. Please add a TODO to follow up and fix this.

Reply:
Actually, that's what I found by digging into [libevent](https://github.com/libevent/libevent/blob/master/listener.c#L179).

Reply:
@mattklein123 just realized I missed your comment. Adding the TODO and yes, this should be pretty easy to customize now, although I'll need a bit of guidance wrt to where we should be configuring this from.

Reply:
A TODO should be good enough for now.

tcp_backlog_size would be a nice future addition in a followup change by either you or someone else.

Reply:
Yup, added it to my TODO list.

Reply:
Can you flesh out the comment more to explain where the magic 128 comes from (libevent)?

Reply:
Done!